src/XCCDF/result.c

```diff
@@ -1067,16 +1080,32 @@ xmlNode *xccdf_target_identifier_to_dom(const struct xccdf_target_identifier *ti
 	}
 }
 
-xmlNode *xccdf_rule_result_to_dom(struct xccdf_rule_result *result, xmlDoc *doc, xmlNode *parent, const struct xccdf_version_info* version_info)
+xmlNode *xccdf_rule_result_to_dom(struct xccdf_rule_result *result, xmlDoc *doc, xmlNode *parent, const struct xccdf_version_info* version_info, struct xccdf_benchmark *benchmark, bool use_stig_rule_id)
 {
+	/* Handle attributes */
+	const char *idref = xccdf_rule_result_get_idref(result);
+	if (benchmark && use_stig_rule_id) {
+		const char *stig_rule_id = NULL;
+		struct xccdf_item *item = xccdf_benchmark_get_member(benchmark, XCCDF_RULE, idref);
+		struct oscap_reference_iterator *references = xccdf_item_get_references(XRULE(item));
+		while (oscap_reference_iterator_has_more(references)) {
+			struct oscap_reference *ref = oscap_reference_iterator_next(references);
+			if (strcmp(ref->href, DISA_STIG_VIEWER_HREF) == 0)
+				stig_rule_id = ref->title;
+		}
+		oscap_reference_iterator_free(references);
+
+		if (stig_rule_id)
+			idref = stig_rule_id;
+		else
+			return;
+	}
+	
 	xmlNs *ns_xccdf = lookup_xccdf_ns(doc, parent, version_info);
 
 	xmlNode *result_node = xmlNewTextChild(parent, ns_xccdf, BAD_CAST "rule-result", NULL);
 
-	/* Handle attributes */
-	const char *idref = xccdf_rule_result_get_idref(result);
-	if (idref)
-		xmlNewProp(result_node, BAD_CAST "idref", BAD_CAST idref);
```

Comment:
this change chances logic, idref was checked before the PR, after the PR it isn't